nightlies/cron_wrapper

```diff
@@ -18,6 +18,13 @@ STDOUT="/var/tmp/stdout.$$"
 # $STDOUT and $STDERR are removed when the script exits for any reason
 trap  "rm -f $STDOUT $STDERR" 0
 
+# lock $LOG with file descriptor 200
+# if $LOC is locked by other process - wait for 420 sec
+exec 200>>$LOG
+# if flock failed, define a new $LOG and keep runing
+flock -w 420 200 || LOG="/var/log/crontab-nightlies-log/crontab_lock_problem.log"
+
+
```

Comment:
ok, that's one of my three points, but  

1) still 420s? 

2) because you're locking before running the command, that means that the entire time the command runs, no one else can write to the log.   Say job A runs for 10s and job B for 419s.  They both start simultaneously, and then job B gets the lock on the shared log; job A could run and complete, but is blocked for the entire runtime of job B.

The solution is to lock only when writing to the shared log, and to drop the lock when not writing.  Since the command itself already outputs to separate files, you can fix this with more invocations of flock.